### PR TITLE
Added favorite button in Performer grid view

### PR DESCRIPTION
**Pull request by CrookedDuck**
Resolves part of  #705
By clicking the favorite icon the user can toggle whether or not the performer is favorite.

### CODE REVIEW

**Comment by kermieisinthehouse** [comment]
Thank your for your contribution! I'd love to merge this but there are a few outstanding comments:

- The state after clicking the icon is ambiguous, it is solid white both after favoriting and unfavoriting. (Tested on Firefox Mobile)
- For the unfavorited state, I believe the shadow should be less dark, and perhaps raise to the current level of darkness on hover.
- There are some lint errors that need to be fixed before CI passes.

Thanks! Feel free to comment with any questions.

**Comment by DogmaDragon** [comment]
Would love a show/hide toggle to be added in the settings. I see it adds a unique class so it can be hidden via CSS, but a toggle would be more user-friendly.

File: ui/v2.5/src/components/Performers/PerformerCard.tsx [hunk 1]
```diff
@@ -61,17 +65,38 @@ export const PerformerCard: React.FC<IPerformerCardProps> = ({
...
 
-  function maybeRenderFavoriteIcon() {
-    if (performer.favorite === false) {
-      return;
-    }
-    return (
-      <div className="favorite">
+  const [updatePerformer] = usePerformerUpdate();
+
+  function maybeRenderFavoriteIcon(){
```
Comment (Teda1) [inline ui/v2.5/src/components/Performers/PerformerCard.tsx:70]: The name of this method can be updated since the render is no longer conditional.

File: .gitignore [hunk 2]
```diff
@@ -63,4 +63,11 @@ node_modules
 
 /stash
 dist
-.DS_Store
\ No newline at end of file
+.DS_Store
```
Comment (Teda1) [inline .gitignore:66]: This commit seems unrelated to your change. Excluding the root the stash directory should have been all that was necessary.

File: ui/v2.5/src/components/Performers/styles.scss [hunk 3]
```diff
@@ -105,6 +113,7 @@
 
 .card {
   &.performer-card {
+
```
Comment (Teda1) [inline ui/v2.5/src/components/Performers/styles.scss:116]: This space is likely what the linter is unhappy with.

File: ui/v2.5/src/components/Performers/styles.scss [hunk 4]
```diff
@@ -85,6 +85,14 @@
     width: 3rem;
   }
 
+  .not-favorite {
+    color: rgba(191, 204, 214, 0.5);
+    filter: drop-shadow(0 0 2px rgba(0, 0, 0, 0.9));
```
Comment (Teda1) [inline ui/v2.5/src/components/Performers/styles.scss:90]: A 50% opacity shadow should be enough here.

**Review by Teda1 — COMMENTED** [review]
The usability of this button on mobile should be improved by removing the hover effects on the bottom. Currently, 2 clicks are required to favorite a performer. 1 click to register the hover effect, and another to actually click the button.

**Comment by Teda1** [comment]
Thanks for your contribution! In the future, it would make tester's lives easier if you don't push changes directly to the `develop` branch.

**Comment by kermieisinthehouse** [comment]
> Thanks for your contribution! In the future, it would make tester's lives easier if you don't push changes directly to the `develop` branch.

I'm not sure I understand what this means?

**Comment by Teda1** [comment]
I essentially had to delete the develop branch I had locally to check out his changes since it was pushed directly to the develop branch on his fork. It's not a huge deal but it was an inconvenience.

**Comment by kermieisinthehouse** [comment]
Ah, I see. I put the following in my .gitconfig,

```
[alias]
        pr = "!f() { if [ $# -lt 1 ]; then echo \"Usage: git pr <id> [<remote>]  # assuming <remote>[=origin] is on GitHub\"; else git checkout -q \"$(git rev-parse --verify HEAD)\" && git fetch -fv \"${2:-origin}\" pull/\"$1\"/head:pr/\"$1\" && git checkout pr/\"$1\"; fi; }; f"
```

So that I can do `git pr 3369` and check out a new branch - then build the docker images from there.

**Comment by Teda1** [comment]
Sounds like that would save me some trouble. I'll need to try it out. Thanks!

**Comment by CrookedDuck** [comment]
> Thanks for your contribution! In the future, it would make tester's lives easier if you don't push changes directly to the `develop` branch.

ill keep in mind to create feature/issue related branches in the future, sorry for the inconvenience ;)

File: ui/v2.5/src/components/Performers/PerformerCard.tsx (hunk 1 above)
Comment (CrookedDuck) [reply ui/v2.5/src/components/Performers/PerformerCard.tsx:70]: totally agree, will update to just 'renderFavoriteIcon'

File: .gitignore (hunk 2 above)
Comment (CrookedDuck) [reply .gitignore:66]: yeah, i kinda didnt want to push my changes to the .gitignore in the first place. ill revert the changes for this pr ;)

**Comment by WithoutPants** [comment]
Fixed up the compile errors and changed the styling to be consistent with the older version. Unfavorited performers now only show the favorite button when the performer card is hovered over, since having all the buttons shown was pretty ugly.
